torch/cuda/graphs.py

```diff
@@ -145,19 +153,36 @@ class graph:
     """  # noqa: B950
 
     default_capture_stream: typing.Optional["torch.cuda.Stream"] = None
+    # Unsure about type here
+    raw_default_capture_stream: typing.Optional["torch.cuda.cudart().cudaStream_t"] = None
 
     def __init__(
         self,
         cuda_graph,
         pool=None,
         stream=None,
         capture_error_mode: str = "global",
+        collect_garbage: bool = True,
     ):
         # Lazy-init of default_capture_stream helps avoid circular-import errors.
-        # Not thread safe, but graphs already have the general (explicitly documented)
-        # restriction that only one capture may be underway at a time in the process.
         if self.__class__.default_capture_stream is None:
-            self.__class__.default_capture_stream = torch.cuda.Stream()
+            # We use an external stream to prevent the rare situation
+            # of a stream that is already in used being grabbed from
+            # the stream pool. This requires destroying the external
```

Comment:
For appropriately rare situations, i would lean towards not supporting until it becomes an actual issue. Or at least have tests for these edge cases to be sure we are correctly handling.

Reply:
Makes sense. I'm inclined to just undo this change to be honest. I'm worried about causing an unexpected behavior or error by stuffing too much into this PR. Developers in the know can simply pass in their own cuda stream torch.cuda.graph() to prevent trying to restart stream capture on the same stream.